sandpack-react/src/utils/sandpackUtils.ts

```diff
@@ -65,45 +70,93 @@ export const getSandpackStateFromProps = (
     }, []);
   }
 
+  // Make sure it resolves the entry file
+  if (!projectSetup.files[projectSetup.entry]) {
+    /* eslint-disable */
+    // @ts-ignore
+    projectSetup.entry = resolveFile(projectSetup.entry, projectSetup.files);
+    /* eslint-enable */
+  }
+
   // If no activePath is specified, use the first open file
   if (!activePath || !projectSetup.files[activePath]) {
-    activePath = projectSetup.main || openPaths[0];
+    activePath = openPaths[0];
   }
 
   // If for whatever reason the active path was not set as open, set it
   if (!openPaths.includes(activePath)) {
     openPaths.push(activePath);
   }
 
-  if (!projectSetup.files[activePath]) {
-    throw new Error(
-      `${activePath} was set as the active file but was not provided`
-    );
-  }
-
   const files = addPackageJSONIfNeeded(
     projectSetup.files,
-    projectSetup.dependencies || {},
-    projectSetup.devDependencies || {},
+    projectSetup.dependencies ?? {},
+    projectSetup.devDependencies ?? {},
     projectSetup.entry
   );
 
-  const environment = projectSetup.environment;
-  const existOpenPath = openPaths.filter((file) => files[file]);
+  const existOpenPath = openPaths.filter((path) => files[path]);
 
-  return { openPaths: existOpenPath, activePath, files, environment };
+  return {
+    openPaths: existOpenPath,
+    activePath,
+    files,
+    environment: projectSetup.environment,
+  };
 };
 
-// The template is predefined (eg: react, vue, vanilla)
-// The setup can overwrite anything from the template (eg: files, dependencies, environment, etc.)
-export const getSetup = (
-  template?: SandpackPredefinedTemplate,
-  inputSetup?: SandpackSetup
-): SandboxTemplate => {
-  // The input setup might have files in the simple form Record<string, string>
-  // so we convert them to the sandbox template format
+export const resolveFile = (
+  path: string,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  files: Record<string, any>
+): string | undefined => {
+  if (!path) return undefined;
 
-  const setup = createSetupFromUserInput(inputSetup);
+  let resolvedPath = undefined;
+
+  let index = 0;
+  const strategies = [".js", ".jsx", ".ts", ".tsx"];
+  const leadingSlash = Object.keys(files).every((file) => file.startsWith("/"));
+
+  while (!resolvedPath && index < strategies.length) {
```

Comment:
This would benefit from some comments for future proofness

Reply:
Yeah, thanks for mentioning it. I'll revisit it later in this [task](https://github.com/codesandbox/sandpack/issues/353), there is still room for improvements there